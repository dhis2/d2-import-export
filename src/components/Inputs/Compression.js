import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_ZIP = { value: 'zip', label: i18n.t('Zip') }
export const OPTION_GZIP = { value: 'gz', label: i18n.t('GZip') }
export const OPTION_NONE = { value: '', label: i18n.t('Uncompressed') }

export const COMPRESSION_KEY = 'compression'
export const COMPRESSION_DEFAULT_VALUE = OPTION_ZIP.value

export const Compression = () => (
    <RadioGroup
        name={COMPRESSION_KEY}
        label={i18n.t('Compression')}
        options={[OPTION_ZIP, OPTION_GZIP, OPTION_NONE]}
        dataTest="input-compression"
    />
)
