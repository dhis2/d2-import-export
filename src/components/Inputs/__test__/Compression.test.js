import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Compression } from '../Compression'

jest.mock('react-final-form')

describe('Input component - Compression', () => {
    it('should render correctly', () => {
        const file = shallow(<Compression />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
