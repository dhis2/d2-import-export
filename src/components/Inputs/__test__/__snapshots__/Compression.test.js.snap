// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Compression should render correctly 1`] = `
<Field>
  <Label>
    Compression
  </Label>
  <RadioGroup
    dataTest="input-compression"
    label="Compression"
    name="compression"
    options={
      Array [
        Object {
          "label": "Zip",
          "value": "zip",
        },
        Object {
          "label": "GZip",
          "value": "gz",
        },
        Object {
          "label": "Uncompressed",
          "value": null,
        },
      ]
    }
  />
</Field>
`;
